### PR TITLE
Physics reactivity for collider component updates and order dependency resolution

### PR DESCRIPTION
**Pull request by AidanCaruso**
## Summary

note: fix these tests

## References
closes #_insert number here_

### CODE REVIEW

File: packages/spatial/src/physics/components/ColliderComponent.tsx [hunk 1]
```diff
@@ -83,13 +83,13 @@ export const ColliderComponent = defineComponent({
...
     const rigidbodyEntity = useAncestorWithComponent(entity, RigidBodyComponent)
-    const rigidbodyComponent = useOptionalComponent(rigidbodyEntity, RigidBodyComponent)
     const physicsWorld = Physics.useWorld(entity)
     const triggerComponent = useOptionalComponent(entity, TriggerComponent)
     const hasCollider = useState(false)
+    const physicsWorldRigidbody = Physics.useWorld(entity)?.Rigidbodies[entity]
 
     useEffect(() => {
-      if (!rigidbodyComponent || !physicsWorld) return
+      if (!rigidbodyEntity || !physicsWorld) return
```
Comment (speigg) [inline packages/spatial/src/physics/components/ColliderComponent.tsx:92]: this needs to be `rigidbodyComponent.value`

File: packages/spatial/src/physics/components/ColliderComponent.tsx [hunk 2]
```diff
@@ -101,7 +101,7 @@ export const ColliderComponent = defineComponent({
         Physics.removeCollider(physicsWorld, entity)
         hasCollider.set(false)
       }
-    }, [physicsWorld, component.shape, rigidbodyEntity, !!rigidbodyComponent, transform.scale])
+    }, [physicsWorld, component.shape, rigidbodyEntity, !!physicsWorldRigidbody, transform.scale])
```
Comment (the-reality-engineer) [inline packages/spatial/src/physics/components/ColliderComponent.tsx:104]: !!physicsWorldRigidbody?.value

**Review by the-reality-engineer — CHANGES_REQUESTED** [review]

File: packages/spatial/src/physics/components/ColliderComponent.tsx (hunk 2 above)
Comment (mrhegemon) [reply packages/spatial/src/physics/components/ColliderComponent.tsx:104]: This doesnt return the wrapped state.

Comment (the-reality-engineer) [reply packages/spatial/src/physics/components/ColliderComponent.tsx:104]: nvm, useWorld() doesn't return a state-wrapped physics world..

**Review by mrhegemon — APPROVED** [review]

**Pull request merged**